src/inference/src/dev/converter_utils.cpp

```diff
@@ -0,0 +1,328 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "converter_utils.hpp"
+
+#include <ie_blob.h>
+#include <ie_common.h>
+#include <ie_compound_blob.h>
+#include <ie_layouts.h>
+
+#include <fstream>
+#include <ie_input_info.hpp>
+#include <ie_version.hpp>
+#include <memory>
+#include <openvino/core/except.hpp>
+#include <openvino/op/parameter.hpp>
+#include <openvino/runtime/exception.hpp>
+#include <openvino/runtime/remote_context.hpp>
+#include <openvino/runtime/tensor.hpp>
+
+#include "any_copy.hpp"
+#include "cnn_network_ngraph_impl.hpp"
+#include "cpp/ie_plugin.hpp"
+#include "cpp_interfaces/interface/ie_iexecutable_network_internal.hpp"
+#include "cpp_interfaces/interface/ie_iplugin_internal.hpp"
+#include "ie_icore.hpp"
+#include "ie_ngraph_utils.hpp"
+#include "openvino/runtime/iplugin.hpp"
+#include "so_ptr.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace {
+
+void fill_input_info(ov::Output<ov::Node>& input, InferenceEngine::InputInfo::Ptr& input_info) {
+    const ov::Output<const ov::Node> const_input(input.get_node(), input.get_index());
+    ov::legacy_convert::fill_input_info(const_input, input_info);
+    auto& rt_info = input.get_rt_info();
+    auto it = rt_info.find("ie_legacy_preproc");
+    if (it != rt_info.end()) {
+        rt_info.erase(it);
+    }
+    it = rt_info.find("ie_legacy_td");
+    if (it != rt_info.end()) {
+        rt_info.erase(it);
+    }
+}
+
+void fill_output_info(ov::Output<ov::Node>& input, InferenceEngine::DataPtr& output_info) {
+    const ov::Output<const ov::Node> const_input(input.get_node(), input.get_index());
+    ov::legacy_convert::fill_output_info(const_input, output_info);
+    auto& rt_info = input.get_rt_info();
+    auto it = rt_info.find("ie_legacy_td");
+    if (it != rt_info.end()) {
+        rt_info.erase(it);
+    }
+}
+
+InferenceEngine::SizeVector get_dims(const ov::Output<const ov::Node>& port,
+                                     const std::function<bool(InferenceEngine::SizeVector& dims)>& callback = {}) {
+    InferenceEngine::SizeVector dims = {};
+    const auto& p_shape = port.get_partial_shape();
+    if (p_shape.is_static())
```

Comment:
We can have dynamic shapes in new API, and need to handle this case if we convert from new API to old

Reply:
No, on this code we can get dynamic shapes (because it can exist inside the Parameter for example).
Later we restore infromation about legacy shapes from the RT info.

Reply:
Will try to remove this code